client/my-sites/stats/stats-date-control/stats-date-control-picker.tsx

```diff
@@ -99,6 +102,7 @@ const DateControlPicker = ( {
 				/>
 				<DateControlPickerShortcuts
 					shortcutList={ shortcutList }
+					currentShortcut={ currentShortcut }
```

Comment:
If we are going to maintain the state in here, then we wouldn't need the `onClick` callback now. IMHO, `currentShortcut` should be more of a `DateControlPickerShortcuts` state, which should allow a `defaultShortcut` to initialize. It then calls back when the value changes via a `onChange` method. Do you think that'll make the responsibilities a bit more clear?

Reply:
That sounds reasonable. I'll have another look and see if we can simplify things in a follow-up PR.